### PR TITLE
Changed testing strategy and added to decimal string method for u128

### PR DESCRIPTION
**Pull request by javiersuweijie**
Tests are written in assemblyscript but run by nodejs as it is easier to verify the results.

### CODE REVIEW

**Comment by MaxGraey** [comment]
I about doing all validation stuffs on AssemblyScript side and print on JS side only status like:
JS:
```js
env: {
   abort(...) { ... },
   printAssertStatus(status: boolean, description: number): void {
      const descString = unpackStringFromBuffer(description);
      console.log((status ? colors.green`PASSED` : colors.red`FAILED`) + ': ' + descString);
   }
}
```
AS:
```ts
declare namespace env {
  function printAssertStatus(status: bool, description: string);
}

@inline
function ok(status: bool, description: string) {
  env.printAssertStatus(status, description);
  assert(status);
}

...

var a = new u128(1);
var b = new u128(-1);
var expect = new u128(0);
ok(a + b == expect, "[u128] One plus negative one should be zero (1 + (-1) == 0)");
```

File: assembly/integer/u128.ts [hunk 1]
```diff
@@ -920,11 +919,55 @@ export class u128 {
...
+    }
+    else if (radix == 10) {
+      return this.doubleDabble();
     }
+    else {
+      return "unknown";
+    }
+  }
 
+  private doubleDabble(): string {
```
Comment (MaxGraey) [inline assembly/integer/u128.ts]: Could you rename and define as `utoa10(value: u128): string` and move to [this file](https://github.com/MaxGraey/bignum.wasm/blob/master/assembly/utils.ts)? I'll improve this implementation later

File: tests/u128.spec.ts [hunk 2]
```diff
@@ -1,53 +1,61 @@
...
-    };
+    var buffer = new Uint8Array(memory.buffer);
+    var imports = buildImports("u128.spec.as", memory, buffer);
 
     WebAssembly.instantiate(file, imports)
         .then((result) => {
-            instance = result.instance.exports;
+            let instance = result.instance.exports;
+            it("should add two numbers", () => {
+                expect(instance['BasicOperationsTest#shouldAddTwoNumbers']()).to.be.eq(1);
```
Comment (MaxGraey) [inline tests/u128.spec.ts]: It seems this `'BasicOperationsTest#shouldAddTwoNumbers'` can be de-mangled and use more prettify. See: https://github.com/AssemblyScript/assemblyscript/blob/master/lib/loader/index.js#L110

File: tests/helloworld.spec.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import { expect } from 'chai';
+import { describe, it } from "mocha";
+
+describe('Sanity check for WebAssembly', () => {
+    it("should be available", () => {
```
Comment (MaxGraey) [inline tests/helloworld.spec.ts]: Plz use two spaced indent everywhere =)

**Comment by javiersuweijie** [comment]
Thanks for the suggestion to use demangle. I can probably auto-generate the tests by iterating through the object. Will work on that and refactor for the next PR.

**Comment by MaxGraey** [comment]
Thanks!
This approach much better.
Add plz tests for comparisons '==', '!=', '<' and etc as a prime test, because all others highly reuse this operators for validation.

File: tests/helloworld.spec.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import { describe, it } from "mocha";
+
+describe('Sanity check for WebAssembly', () => {
+  it("should be available", () => {
+    expect(WebAssembly.Module).to.be.not.eq(undefined);
+  });
+});
+
+/*
+describe('Basic u128 operations', () => {
```
Comment (MaxGraey) [inline tests/helloworld.spec.ts]: Clean up this old stuffs plz

**Comment by MaxGraey** [comment]
LGTM. Will there be any further additions?

**Comment by javiersuweijie** [comment]
Please merge this if its okay! I'm thinking of implementing division with a
simple shift and subtract next.

**Comment by MaxGraey** [comment]
Thanks!

**Pull request merged**